docs/02-developers/04-quickstart/hardhat-ignition.md

```diff
@@ -0,0 +1,377 @@
+```
+
+:::info[A breakdown of each part of the Lock Module:]
+
+ - `m.getParameter`: This retrieves a deployment parameter, allowing for a default value to be specified if none is provided.  
+ 
+  **Constants**:  
+  * `JAN_1ST_2030`: The Unix timestamp for the unlock time.  
+  * `ONE_GWEI`: The value of 1 Gwei, expressed in Wei.
+
+:::
+
+
+
+  </Step>
+  <Step title="Set Up Environment Variables">
+    1. Create a file named .env in the root folder of your project.
+
+2. Add the following lines to the file:
+
+```
+RSK_MAINNET_RPC_URL=<your-mainnet-rpc-url>
+RSK_TESTNET_RPC_URL=<your-testnet-rpc-url>
+PRIVATE_KEY=<your-deployer-account-private-key>
+```
+
+:::info[What These Mean]
+
+   * `RSK_MAINNET_RPC_URL`: This connects you to the Rootstock Mainnet (real transactions).  
+   * `RSK_TESTNET_RPC_URL`: This connects you to the Rootstock Testnet (fake money for testing).  
+   * `PRIVATE_KEY`: This is like your account password but in a very secure format.  
+ 
+ **How to Get These**:
+
+   * Visit the [Rootstock RPC API](https://dev.rootstock.io/developers/rpc-api/rootstock/) to get the **Mainnet** or **Testnet** URLs.  
+   * Get your account’s private key from your wallet (e.g., **Metamask**).  
+   * For Testnet tokens, go to the [Rootstock Faucet](https://faucet.rsk.co/).
+
+:::
+ 
+  </Step>
+   <Step title="Compile and Test Your Smart Contracts"> 
+   - Run this command to compile the Contract:
+
+```
+npx hardhat compile
+```
+
+This checks for errors and prepares your contract for deployment.
+
+ 
+- Run this command to check if the contracts behave as expected:
+
+```
+npx hardhat test
+```
+
+3. If everything is okay, you’ll see green checkmarks or messages saying the tests passed.
+
+```
+Generating typings for: 2 artifacts in dir: typechain-types for target: ethers-v6
+Successfully generated 8 typings!
+Compiled 2 Solidity files successfully (evm target: paris).
+
+
+  Box
+    Deployment
+      ✔ Should initialize with a value of 0 (1214ms)
+    Store and Retrieve
+      ✔ Should store the value and retrieve it correctly
+      ✔ Should emit a ValueChanged event on storing a value
+
+  Lock
+    Deployment
+      ✔ Should set the right unlockTime
+      ✔ Should set the right owner
+      ✔ Should receive and store the funds to lock
+      ✔ Should fail if the unlockTime is not in the future
+    Withdrawals
+      Validations
+        ✔ Should revert with the right error if called too soon
+        ✔ Should revert with the right error if called from another account
+        ✔ Shouldn't fail if the unlockTime has arrived and the owner calls it
+      Events
+        ✔ Should emit an event on withdrawals
+      Transfers
+        ✔ Should transfer the funds to the owner
+
+
+  12 passing (1s)
+
+```
+  </Step>
+   <Step title="Deploy the Smart Contract">
+    The following command is used to deploy a smart contract to the Rootstock Testnet:
+
+```
+npx hardhat ignition deploy --network rskTestnet ignition/modules/Box.ts
```

Comment:
Got this error:

```
owanate@Owanates-MacBook-Pro rootstock-hardhat-ignition-starterkit % npx hardhat ignition deploy --network rskTestnet ignition/modules/Box.ts
✔ Confirm deploy to network rskTestnet (31)? … yes
Hardhat Ignition 🚀

Deploying [ BoxModule ]

Batch #1
  Executing BoxModule#Box...

An unexpected error occurred:

IgnitionError: IGN401: Error while executing BoxModule#Box: all the transactions of its network interaction 1 were dropped. Please try rerunning Hardhat Ignition.
```

Reply:
Deleted artifacts and it worked. Capture this in the troubleshooting tip at the bottom or here

Reply:
It was already added earlier.

Reply:
I updated the content with a warning to delete the folder at the top and also added this error too with same warning incase someone misses it.